api/tests/unit/domain/usecases/retrieve-last-or-create-certification-course_test.js

```diff
@@ -1,331 +1,296 @@
+          sinon.assert.notCalled(certificationCourseRepository.save);
+          sinon.assert.notCalled(assessmentRepository.save);
+          sinon.assert.notCalled(certificationChallengeRepository.save);
         });
       });
 
-      context('when the user has a link with a certification candidate in the session', () => {
-
-        it('should create the certification course with status "started", if at least 5 competences with level higher than 0', async function() {
-          // given
-          sinon.stub(certificationCandidateRepository, 'getBySessionIdAndUserId')
-            .resolves({ firstName: 'prénom', lastName: 'nom', birthdate:'DDN', birthplace:'lieu' });
-          const certificationProfile = new CertificationProfile({ userCompetences: fiveCompetencesWithLevelHigherThan0 });
-          sinon.stub(userService, 'getCertificationProfile').withArgs({ userId, limitDate: now })
-            .resolves(certificationProfile);
-          sinon.stub(userService, 'fillCertificationProfileWithCertificationChallenges').withArgs(certificationProfile)
-            .resolves(certificationProfile);
-          sinon.stub(certificationChallengesService, 'saveChallenges').resolves(certificationCourseWithNbOfChallenges);
-          sinon.stub(certificationCourseRepository, 'save').resolves(certificationCourse);
-          sinon.stub(assessmentRepository, 'save').resolves();
+      context('when user is certifiable', () => {
 
-          // when
-          const newCertification = await retrieveLastOrCreateCertificationCourse({
-            sessionId,
-            accessCode,
-            userId,
-            sessionRepository,
-            userService,
-            certificationCandidateRepository,
-            certificationChallengesService,
-            certificationCourseRepository,
-            assessmentRepository,
+        beforeEach(() => {
+          certificationProfile = { isCertifiable: sinon.stub().returns(true), userCompetences: 'someUserCompetences' };
+          userService.getCertificationProfile.withArgs({ userId, limitDate: now }).resolves(certificationProfile);
+        });
+
+        context('when a certification course has been created meanwhile', () => {
+          
+          let existingCertificationCourse;
+          beforeEach(() => {
+            existingCertificationCourse = Symbol('existingCertificationCourse');
+            certificationCourseRepository.getLastCertificationCourseByUserIdAndSessionId
+              .withArgs(userId, sessionId).onCall(1).resolves(existingCertificationCourse);
+            userService.fillCertificationProfileWithCertificationChallenges.withArgs(certificationProfile).resolves();
           });
 
-          // then
-          expect(newCertification).to.deep.equal({
-            created: true,
-            certificationCourse: certificationCourseWithNbOfChallenges
+          it('should return it with flag created marked as false', async function() {
```

Comment:
Peut être que c'est juste moi mais, idem ici je pense qu'on peut expliciter le 'it'

Reply:
Noté